opentelemetry/src/global/logs.rs

```diff
@@ -25,6 +26,9 @@ pub trait ObjectSafeLoggerProvider {
         &self,
         library: Arc<InstrumentationLibrary>,
     ) -> Box<dyn Logger + Send + Sync + 'static>;
+
+    /// shutdown the logger provider, logs emitted after this will not be processed.
```

Comment:
Was thinking about this a bit more, specifically for logging.
It makes sense to have these Global things for metrics, traces, so that application can retrive the global provider and then get tracers,meters from them.

But for logs - these is no such use case. LoggerProviders are *not* meant for end users to get loggers from. It is only required for the log bridges. 

The expected usage pattern shown here : https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-appender-tracing/examples/basic.rs

It works totally fine as-is. The user creates provider and provides a ref to it, to the bridges, but retains ownership. And they can continue to keep ownership and call shutdown at the end of the application itself. What scenario requires setting the global provider for logging?

@TommyCpp @lalitb 
Sorry for this question after agreeing with the direction of this PR overall :(
I think we should treat logs differently from other signals. The idea shown in this PR can be used for other signals, but not for logs...

Please share your thoughts..

Reply:
It works totally fine as-is.

Then we should remove shutdown_logger_provider from global module. Otherwise users will just try to call it without actually shutting down anything

Reply:
Yes and yes.
We need to keep shutdown only for the sdk!

Reply:
>  What scenario requires setting the global provider for logging?

While this repo provides/maintains the tokio-tracing logging bridge, which doesn't need setting global provider, the bridge-api should be extensible enough for users to create their appenders for some other logging API. We can't assume the requirements of that appender, and in that case good to provide the support recommended by [API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#loggerprovider):
```
Thus, the API SHOULD provide a way to set/register and access a global default LoggerProvider.
```

Reply:
Regarding shutdown_signal_provider, I believe it should be removed from global for all signals, as it gives an impression that these can be used by instrumentation libraries. We have an issue for that.

Reply:
> > What scenario requires setting the global provider for logging?
> 
> While this repo provides/maintains the tokio-tracing logging bridge, which doesn't need setting global provider, the bridge-api should be extensible enough for users to create their appenders for any logging framework. We can't assume the requirements of that appender, and in that case good to provide the support recommended by [API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md#loggerprovider):
> 
> ```
> Thus, the API SHOULD provide a way to set/register and access a global default LoggerProvider.
> ```

Both the appenders we have be hosting in this repo did not require global::logger, and it works by accepting LoggerProvider in its new() method. This can be used by other appenders as well.

(If there is good demand, we can expose it, but I'd suggest to keep the public API _as minimal as possible_ for initial stable release.)